puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/foreman_plugin.groovy

```diff
@@ -0,0 +1,121 @@
+pipeline {
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    agent none
+
+    stages {
+        stage('Test') {
+            matrix {
+                agent { label 'fast' }
+                axes {
+                    axis {
+                        name 'ruby'
+                        values '2.5', '2.6', '2.7'
+                    }
+                }
+                stages {
+                    stage('Configure Environment') {
+                        steps {
+                            configureRVM(ruby)
+                            deleteDir()
+                            checkout scm: [
+                                    $class: 'GitSCM',
+                                    branches: [[name: params.plugin_branch]],
+                                    userRemoteConfigs: [[url: params.plugin_repo]],
+                                    extensions: [[$class: 'CloneOption', shallow: true, noTags: true]]
+                                ],
+                                poll: false
+
+                            dir('foreman') {
+                                checkout scm: [
+                                        $class: 'GitSCM',
+                                        branches: [[name: params.foreman_branch]],
+                                        userRemoteConfigs: [[url: params.foreman_repo]],
+                                        extensions: [[$class: 'CloneOption', shallow: true, noTags: true]]
+                                    ],
+                                    poll: false
+
+                                addGem()
+                                databaseFile(gemset())
+                                configureDatabase(ruby)
+                            }
+                        }
+                    }
+                    stage('Unit Tests') {
+                        steps {
+                            dir('foreman') {
+                                withRVM(['bundle exec rake jenkins:unit TESTOPTS="-v" --trace'], ruby)
```

Comment:
Is this really the way to test a plugin?

Reply:
I think this will run all tests (including foreman core ones).
For my plugin, I test with `rake test:foreman_rh_cloud`.

Reply:
That's exactly my point, but I think currently this is how we test in Jenkins since I copied the builder.

Reply:
So maybe it makes sense to make core tests at least optional? So I will be able to decide for my plugin if I want to run core tests too. For smaller plugins that do not modify core objects it should be OK to run only plugin related tests.

Reply:
This is still a draft for now, so this line was partly a TODO so I wouldn't forget.